pkg/volume/glusterfs/glusterfs.go

```diff
@@ -98,15 +98,17 @@ func (plugin *glusterfsPlugin) GetPluginName() string {
 }
 
 func (plugin *glusterfsPlugin) GetVolumeName(spec *volume.Spec) (string, error) {
-	volumeSource, _, err := getVolumeSource(spec)
+	var endpointName string
+
+	if spec.Volume != nil && spec.Volume.Glusterfs != nil {
```

Comment:
this doesn't appear to handle PV volumes correctly

Reply:
Added pv volumes to the func.